### ISSUE TITLE
tsconfig.json doesn't appear to have any affect.

### ISSUE THREAD

**Issue by vietnogi** [janaagaard75/expo-and-typescript#25]
Can someone verify that changes to tsconfig.json have an affect.  If so, what was changed?

**Comment by janaagaard75** [comment]
You're absolutely right. Thanks a lot for pointing this out - I had not noticed this.

I've looked a bit into this, and here's the deal: The compilation used to be done by `react-native-typescript-transformer`, but [Since version 31, Expo ships with Babel 7](https://blog.expo.io/expo-sdk-v31-0-0-is-now-available-cad6d0463f49), and Babel is now used to compile TypeScript into JavaScript. Well, Babel doesn't really compile the TypeScript code, it merely [strips the type annotations](https://iamturns.com/typescript-babel/) (and probably a few other things too).

TypeScript's `tsc` and `tsconfig.json` are now only used to verify the code source. This can be done with the `validate` run script, e.g. `yarn validate` or `npm run validate`.

TL;DR: You now have to run `yarn validate` to check your code.

**Comment by vietnogi** [comment]
Yea! I figured this out after couple of days O_O.  Here is the solution that I was trying to solve: https://github.com/janaagaard75/expo-and-typescript/issues/21#issuecomment-456976812
